tools/fapi/tss2_gettpm2object.c

```diff
@@ -0,0 +1,202 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <tss2/tss2_tctildr.h>
+#include <tss2/tss2_esys.h>
+#include <tss2/tss2_mu.h>
+
+#include "tools/fapi/tss2_template.h"
+#include "lib/tpm2.h"
+#include "lib/files.h"
+
+/* Context struct used to store passed commandline parameters */
+static struct cxt {
+    char const *data;
+    char const *path;
+    char const *tcti;
+    bool        overwrite;
+} ctx;
+
+/* Parse commandline parameters */
+static bool on_option(char key, char *value) {
+    switch (key) {
+    case 'o':
+        ctx.data = value;
+        break;
+    case 'f':
+        ctx.overwrite = true;
+        break;
+    case 'p':
+        ctx.path = value;
+        break;
+    case 'T':
+        ctx.tcti = value;
+        break;
+    }
+    return true;
+}
+
+/* Define possible commandline parameters */
+static bool tss2_tool_onstart(tpm2_options **opts) {
+    struct option topts[] = {
+        {"path", required_argument, NULL, 'p'},
+        {"tpm2object", required_argument, NULL, 'o'},
+        {"force" , no_argument, NULL, 'f'},
+        {"tcti", required_argument, NULL, 'T'},
+
+    };
+    return (*opts = tpm2_options_new ("o:fp:", ARRAY_LEN(topts), topts,
+                                      on_option, NULL, 0)) != NULL;
+}
+
+/* Execute specific tool */
+static int tss2_tool_onrun (FAPI_CONTEXT *fctx) {
+    uint8_t blob_type;
+    uint8_t *esys_blob = NULL;
+    size_t esys_blob_size;
+    ESYS_CONTEXT *esys_ctx = NULL;
+    ESYS_TR esys_handle = ESYS_TR_NONE;
+    ESYS_TR esys_handle_deser = ESYS_TR_NONE;
+    FILE *stream = NULL;
+    TPM2_HANDLE tpm_handle;
+    TSS2_RC e_rc;
+    TSS2_TCTI_CONTEXT *tcti = NULL;
+    const char *tcti_conf_option = NULL;
+
+    tool_rc t_rc;
+
+    /* Check availability of required parameters */
+    if (!ctx.path) {
+        fprintf (stderr, "path is missing, use --path\n");
+        return -1;
+    }
+
+    if (!ctx.data) {
+        ctx.data = "-";
+    }
+
+    /* Execute FAPI command with passed arguments */
+    TSS2_RC r = Fapi_GetEsysBlob (fctx, ctx.path, &blob_type, &esys_blob, &esys_blob_size);
+    if (r != TSS2_RC_SUCCESS) {
+        LOG_PERR ("Fapi_GetAppData", r);
+        return 1;
+    }
+
+    if (strcmp(ctx.data, "-")) {
+        if (!ctx.overwrite) {
+            FILE *fp = fopen(ctx.data, "rb");
+            if (fp) {
+                fclose(fp);
+                LOG_ERR("Path: %s already exists. Please rename or delete the file!\n",
+                ctx.data);
+                goto error;
+            }
+        }
```

Comment:
isn't this a time of check time of use bug... You would really want to use the exclusive flag. I know all the libc's I've used support the "x" flag for that.

I don't even really know why we need this force option. If someone gives you an output path, just truncate and write it.

Reply:
> isn't this a time of check time of use bug... You would really want to use the exclusive flag. I know all the libc's I've used support the "x" flag for that.
> 
> I don't even really know why we need this force option. If someone gives you an output path, just truncate and write it.

I also do not know why we need this option and could remove it ;-)